docs/proposals/cron-wf-improvement-proposal.md

```diff
@@ -0,0 +1,102 @@
+
+- maxFailures - maximum number of **consecutive failures** before stopping the scheduling of a workflow
+
+- maxSuccess - maximum number of workflows with success.
+
+## How to store state
+
+Since we need to control how many child workflows had success/failure we must store state. With this some questions arise:
+
+- Should we just store it through the lifetime of the controller or should we store it to a database?
+
+    - Probably only makes sense if we can backup the state somewhere (like a BD). However, I don't have enough knowledge about workflow's architecture to tell how good of an idea this is.
+- If a CronWorkflow gets re-applied, does it maintain or reset the number of success/failures?
+
+    - I guess it should reset since a configuration change should be seen as a new start.
+
+## How to stop the workflow
+
+Once the configured number of failures or successes is reached, it is necessary to stop the workflow scheduling.
+I believe we have 3 options:
+
+- Delete the workflow: In my opinion, this is the worst option and goes against gitops principles.
+- Suspend it (set suspend=true): the workflow spec is changed to have the workflow suspended. I may be wrong but this conflicts with gitops as well.
+- Stop scheduling it: The workflow spec is the same. The controller needs to check if the max number of runs was already attained and skip scheduling if it did.
+
+Option 3 seems to be the only possibility. After reaching the max configured executions, the cron workflow would exist forever but never scheduled. Maybe we could add a new status field, like `Inactive` and have something the UI to show it?
+
+## How to handle suspended workflows
+
+One possible case that comes to mind is a long outage where all workflows are failing. For example, imagine a workflow that needs to download a file from some storage and for some reason that storage is down. Workflows will keep getting scheduled but they are going to fail. If they fail the number of configured `maxFailures`, the workflows gets stopped forever. Once the storage is back up, how can the user enable the workflow again?
+
+- Manually re-create the workflow: could be an issue if the user has multiple cron workflows
+- Instead of stopping the workflow scheduling, introduce a back-off period as suggested by [#7291](https://github.com/argoproj/argo-workflows/issues/7291). Or maybe allow both configurations.
+
+I believe option 2 would allow the user to select if they want to stop scheduling or not. If they do, when cron workflows are wrongfully halted, they will need to manually start them again. If they don't, Argo will only introduce a back-off period between schedules to avoid rescheduling workflows that are just going to fail. Spec could look something like:
```

Comment:
I agree with this, but I would say that we should probably ship this feature in a separate PR than the above requirements you outlined. It feels like this would add additional complexity.